clients/fides-js/src/lib/events.ts

```diff
@@ -8,12 +8,11 @@ declare global {
 
 /**
  * Defines the type of "extra" details that can be optionally added to certain
- * events. This is intentionally vague, but constrained to be basic (primitive)
- * values for simplicity.
+ * events. This is intentionally vague. See the /docs/fides-event.ts
  */
 export type FidesEventExtraDetails = Record<
   string,
-  string | number | boolean | undefined
+  string | number | boolean | object | undefined
```

Comment:
In TypeScript, using object as a type is generally discouraged because it's too broad and can lead to type safety issues. The object type includes everything except primitives (string, number, boolean, symbol, null, undefined), which means it could accept arrays, functions, and other complex types that might not be what you expect. Unless that _is_ what we're expecting and that's part of the "vague for simplicity" we're talking about?

```suggestion
  string | number | boolean | Record<string, unknown> | undefined
```

Reply:
Hah, I swear I looked that up. Thank you!